src/python/pants/backend/python/target_types.py

```diff
@@ -724,6 +725,93 @@ def compute_value(cls, raw_value: Optional[PythonArtifact], address: Address) ->
         return cast(PythonArtifact, super().compute_value(raw_value, address))
 
 
+class PythonDistributionEntryPoints(NestedDictStringToStringField, AsyncFieldMixin):
+    alias = "entry_points"
+    required = False
+    help = (
+        "Any entry points, such as `console_scripts` and `gui_scripts`.\n"
+        "\n"
+        "Specify as a nested dictionary, with a dictionary for each type of entry point, "
+        "e.g. `console_scripts` vs. `gui_scripts`. Each dictionary maps the entry point name to "
+        'either a setuptools entry point ("path.to.module:func") or a Pants target address to a '
+        "pex_binary target.\n\n"
+        + dedent(
+            """\
+            Example:
+
+                entry_points={
+                  "console_scripts": {
+                    "my-script": "project.app:main",
+                    "another-script": "project/subdir:pex_binary_tgt"
+                  }
+                }
+
+            """
+        )
+        + "Note that Pants will assume that any value that either starts with `:` or has `/` in it, "
+        "is a target address to a pex_binary target. Otherwise, it will assume it's a setuptools "
+        "entry point as defined by "
+        "https://packaging.python.org/specifications/entry-points/#entry-points-specification. Use "
+        "`//` as a prefix for target addresses if you need to disambiguate.\n\n"
+        + dedent(
+            """\
+            Pants will attempt to infer dependencies, which you can confirm by running:
+
+                ./pants dependencies <python_distribution target address>
+
+            """
+        )
+    )
+
+
+@dataclass(frozen=True)
+class PythonDistributionPexableEntryPoint:
+    """Pexable means that the entry point could be from a pex_binary target."""
+
+    entry_point: EntryPoint
+    pex_binary_address: Optional[Address]
+
+
+# See `target_type_rules.py` for the `Resolve..Request -> Resolved..` rule
+@dataclass(frozen=True)
+class ResolvedPythonDistributionEntryPoints:
+    val: FrozenDict[str, FrozenDict[str, PythonDistributionPexableEntryPoint]] = FrozenDict()
```

Comment:
I forgot what this was supposed to be. This might make it more clear:

```suggestion
    categories_to_entry_points: FrozenDict[str, FrozenDict[str, PythonDistributionPexableEntryPoint]] = FrozenDict()
```

Maybe add a comment: "E.g. `{"console_scripts": {"ep": PythonDistributionEntryPoint(...)}}`.